lms/templates/courseware/submission_history.html

```diff
@@ -1,13 +1,14 @@
+<%page expression_filter="h"/>
 <% import json  %>
-<h3>${username | h} > ${course_id | h} > ${location | h}</h3>
+<h3>${username} > ${course_id} > ${location}</h3>
 
 % for i, (entry, score) in enumerate(zip(history_entries, scores)):
 <hr/>
 <div>
 <b>#${len(history_entries) - i}</b>: ${entry.updated} (${TIME_ZONE} time)</br>
 Score: ${score.grade} / ${score.max_grade}
 <pre>
-${json.dumps(entry.state, indent=2, sort_keys=True) | h}
+${json.dumps(entry.state, indent=2, sort_keys=True)}
```

Comment:
Yeah - it's a super-niche page (this is Show Submissions in the LMS courseware).  So it's intended for course teams / support to use for debugging, and it's effectively just a dump of the database table as JSON.